src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -35,6 +35,8 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
     {
         protected abstract bool CanOffer(SyntaxNode body);
         protected abstract bool PrefersThrowExpression(DocumentOptionSet options);
+        protected abstract SyntaxToken CreateInterpolatedStringStartToken();
```

Comment:
Maybe move these to SyntaxGenerator, and keep everything here static?
Then you could also remove the abstract methods from AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs which seems to be the other user of this generator API.

Reply:
This is good idea!